### PR TITLE
use NAN for Node.js >=0.12 compatibility of native add-on

### PR DESCRIPTION
**Pull request by not-implemented**
I've ported the binding.cc to NAN to be compatible with all supported node versions.

### CODE REVIEW

File: binding.cc [hunk 1]
```diff
@@ -1,62 +1,65 @@
...
 static VCHI_INSTANCE_T vchi;
 static VCHI_CONNECTION_T *vchi_connections;
 
 static char buf[1024];
 
-Handle<Value> Request(const Arguments& args) {
-    HandleScope scope;
+NAN_METHOD(Request) {
+    if (info.Length() < 1) {
+        Nan::ThrowTypeError("Wrong number of arguments");
```
Comment (loyd) [inline binding.cc]: As far as I am concerned, while `request()` is internal, here must be `assert()`

File: binding.cc [hunk 2]
```diff
@@ -1,62 +1,65 @@
...
+        return;
+    }
+    if (!info[0]->IsString()) {
+        Nan::ThrowTypeError("First argument must be a string");
```
Comment (loyd) [inline binding.cc]: And here.

File: binding.cc [hunk 3]
```diff
@@ -1,62 +1,65 @@
...
+        return;
+    }
 
-    String::AsciiValue str(args[0]);
+    v8::String::Utf8Value str(info[0]);
```
Comment (loyd) [inline binding.cc:14]: Can `vcgencmd` return non-ascii?

**Comment by loyd** [comment]
In general, thanks.

File: binding.cc (hunk 1 above)
Comment (not-implemented) [inline binding.cc]: Ok yes, I'll change that.

File: binding.cc (hunk 3 above)
Comment (not-implemented) [inline binding.cc:14]: v8::String::AsciiValue seems to be deprecated according to https://strongloop.com/strongblog/node-js-v0-12-c-apis-breaking/ "Removal of v8::String::AsciiValue" ...

I don't know if Utf8Value is the best choice here, but as the commands are all ASCII, it doesn't matter.

**Comment by not-implemented** [comment]
Updated branch with assert() now

**Comment by not-implemented** [comment]
Cool, tnx for the new version :-)

**Pull request merged**